utils/schemed-kafka-components-utils/src/test/scala/pl/touk/nussknacker/engine/schemedkafka/encode/BestEffortJsonSchemaEncoderTest.scala

```diff
@@ -258,6 +258,25 @@ class BestEffortJsonSchemaEncoderTest extends AnyFunSuite {
     new BestEffortJsonSchemaEncoder(ValidationMode.lax).encode(Map("foo" -> null), schema) shouldBe 'valid
   }
 
+  test("should encode union") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "object",
+        |  "properties": {
+        |    "foo": {
+        |      "type": ["string", "integer"]
```

Comment:
Here is different union presentation way then above. I mean "anyOf".

Reply:
added all three possibilities to define union, that we support when typing (which is covered in tests)